Getting started docs (take 2)

cjrace opened the pull request:
# Brief overview of changes

Couple of quick changes to move the workflow vignette to a getting started page and to make the readme point users to it.

## Why are these changes being made?

Make it easier for new users to get to the workflow page, it's where we want to funnel them to.

## Detailed description of changes

Naming a vignette as the package name makes it a getting started page, I did that as our workflow vignette is basically that!

## Additional information for reviewers

Looks like this now
![image](https://github.com/user-attachments/assets/304743c8-fd56-4abe-9b69-b68b1b38fe47)

## Issue ticket number/s and link

Resolves #57 

Replaces #59 - that had conflicts, so I rebased and then rather than remove the origin version of the branch, I just made a new branch locally and pushed that as this.

rmbielby reviewed (changes requested):
Looks like there's been a change to the publication list since you created the PR as the `get_publications()` check is failing for me here, so maybe worth just running a quick `seed_tests()` again. Would be good once we've added the search functionality to `get_publications()` (#61) if the test is limited to a subset of publications based on a fairly non-common search term, such that it's not changing the results as often.

<img width="220" alt="image" src="https://github.com/user-attachments/assets/f6f445b6-ab9f-4ca5-838e-b8fc30f01b55">

Also, assuming that this is the final change before testing starts, can we turn the release number over to 0.3.2 in this PR so we've got a clear release version with which we're starting testing?

cjrace commented:
> Looks like there's been a change to the publication list since you created the PR as the `get_publications()` check is failing for me here, so maybe worth just running a quick `seed_tests()` again. Would be good once we've added the search functionality to `get_publications()` (#61) if the test is limited to a subset of publications based on a fairly non-common search term, such that it's not changing the results as often.
> 
> <img alt="image" width="220" src="https://private-user-images.githubusercontent.com/230859/380145781-f6f445b6-ab9f-4ca5-838e-b8fc30f01b55.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4NjAyNTMsIm5iZiI6MTcyOTg1OTk1MywicGF0aCI6Ii8yMzA4NTkvMzgwMTQ1NzgxLWY2ZjQ0NWI2LWFiOWYtNGNhNS04MzhlLWI4ZmMzMGYwMWI1NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyNVQxMjM5MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMjkxN2FmMzM3ZGMzNjJkNjlmNTJhYTIxMjIwYjkyYjk4ZGE1ZTZhMDBjZWE4NzhjMjdhMjM2ZjM2YTljZWNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.VI3jCnKt3HWrKYPHT2F_iApz07iCQGboX8D7j7u914Y">
> Also, assuming that this is the final change before testing starts, can we turn the release number over to 0.3.2 in this PR so we've got a clear release version with which we're starting testing?

Good points, have re-seeded the test data and bumped the version to 0.4.0 (felt a minor change was probably more fitting given we did move all the example IDs, rename at least one function and changed the default behaviours around a little.
